src/codegen.md

```diff
@@ -0,0 +1,64 @@
+# Code generation
+
+<!-- TODO: target_feature -->
+
+## Optimization hints
+
+The `cold` and `inline` [attributes] give suggestions to the compiler to
+compile your code in a way that may be faster than what it would do without
+the hint. The attributes are only suggestions, and the compiler may choose to
+ignore it.
+
+Both attributes can be used on closures and [functions]. When applied to a
+function in a [trait], they apply only to that function when used as a default
+function for a trait implementation and not to all trait implementations. The
+attributes have no effect on a trait function without a body.
+
+<!-- TODO: I believe it is currently not possible to use these on closures
```

Comment:
I can't find a way to get the cold attribute onto a closure. So removing it seems fine. Do it in it its own PR after this one though (or file an issue)